test/evp_test.c

```diff
@@ -2516,13 +2499,23 @@ static int parse(EVP_TEST *t)
         }
         OPENSSL_free(keybin);
     }
-    if (pkey != NULL && key_disabled(pkey)) {
-        EVP_PKEY_free(pkey);
-        pkey = NULL;
-    }
 
     /* If we have a key add to list */
     if (klist != NULL) {
+        if (t->s.numpairs != 1)
+            TEST_info("Line %d: missing blank line\n", t->s.curr);
+
+        /* Hack to detect SM2 keys */
+        if(strstr(pp->value, "SM2")) {
+#ifdef OPENSSL_NO_SM2
```

Comment:
This SM2 specific code is there in your patch too? I just moved it.

Or are you referring specifically to the `OPENSSL_NO_SM2` guard? If so, I think it is unavoidable. Before it was still there - but was inside the key_disabled() function. I've just removed that function and moved the guard here instead. Since this is in SM2 specific code anyway, having an `OPENSSL_NO_SM2` guard doesn't seem unreasonable to me.